### PR TITLE
Prf #190

### LINKED ISSUE #190: Implement FieldElement prf
Need an implementation of a prf that generates random field elements. In particular, need to an implementation for the following interface: [fresco/tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/utils/Prf.java](https://github.com/aicis/fresco/blob/mascot-preprocessing/tools/mascot/src/main/java/dk/alexandra/fresco/tools/mascot/utils/Prf.java).

### PR DESCRIPTION
**Pull request by GuutBoy**
Introduces a Deterministic Random Number Generator (DRNG) interface, to deterministicly generate random numbers in a given range. By picking the range to be _0, ... , p_ for a prime modulus _p_ this allows us to generate elements in some _Zp_, thus this fixes #190.

The implementation is based on the Deterministic Random Bit Generator (DRBG) interface using rejection sampling. Also included is a DRBG based on AES in counter mode.

### CODE REVIEW

File: core/src/main/java/dk/alexandra/fresco/framework/util/AesCtrDrbg.java [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import javax.crypto.ShortBufferException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Implementation of a DRBG using AES in counter mode.
+ */
+public class AesCtrDrbg implements Drbg {
+
+  private Cipher cipher;
```
Comment (pffrandsen) [inline core/src/main/java/dk/alexandra/fresco/framework/util/AesCtrDrbg.java]: The order of declaration can be improved, let the static fields come first.

File: core/src/main/java/dk/alexandra/fresco/framework/util/AesCtrDrbg.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
+package dk.alexandra.fresco.framework.util;
+
+import java.security.GeneralSecurityException;
+import javax.crypto.Cipher;
+import javax.crypto.ShortBufferException;
+import javax.crypto.spec.IvParameterSpec;
+import javax.crypto.spec.SecretKeySpec;
+
+/**
+ * Implementation of a DRBG using AES in counter mode.
```
Comment (pffrandsen) [inline core/src/main/java/dk/alexandra/fresco/framework/util/AesCtrDrbg.java]: Write the DBRG abbreviation out in the javadoc

File: core/src/test/java/dk/alexandra/fresco/framework/util/TestAesCtrDrbg.java [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  public void testAesCtrDrbg() {
+    Random rand = new Random(42);
+    byte[] seed = new byte[32];
+    rand.nextBytes(seed);
+    new AesCtrDrbg(seed);
+
+    try {
+      new AesCtrDrbg(new byte[33]);
+      fail("Constructor fail to throw exception on too long seed");
+    } catch (IllegalArgumentException e) {
```
Comment (pffrandsen) [inline core/src/test/java/dk/alexandra/fresco/framework/util/TestAesCtrDrbg.java]: You can expect exceptions which would improve readability here

I.e. Test(expected=IllegalArgumentException.class)

File: core/src/test/java/dk/alexandra/fresco/framework/util/TestAesCtrDrbg.java [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+
+  @Test
+  public void testInitCipher() {
+    Random rand = new Random(42);
+    byte[] seed = new byte[32];
+    rand.nextBytes(seed);
+    AesCtrDrbg drbg = new AesCtrDrbg(seed);
+    try {
+      drbg.initCipher(new byte[16], new byte[16]);
+    } catch (Exception e) {
```
Comment (pffrandsen) [inline core/src/test/java/dk/alexandra/fresco/framework/util/TestAesCtrDrbg.java]: Dont catch an exception, the unittest provides this functionality directly

File: core/src/test/java/dk/alexandra/fresco/framework/util/TestAesCtrDrbg.java (hunk 3 above)
Comment (GuutBoy) [reply core/src/test/java/dk/alexandra/fresco/framework/util/TestAesCtrDrbg.java]: Sure, but then it would not be certain that all exceptions are thrown.

**Pull request merged**